DigitalLearningSolutions.Web/Controllers/LearningPortalController/Current.cs

```diff
@@ -220,5 +286,22 @@ public IActionResult RemoveResourceFromActionPlanPost(int learningLogItemId)
             actionPlanService.RemoveActionPlanResource(learningLogItemId, User.GetCandidateIdKnownNotNull());
             return RedirectToAction("Current");
         }
+
+        public bool IsDateBlank(int? day, int? month, int? year)
+        {
+            return (day ?? 0) == 0 && (month ?? 0) == 0 && (year ?? 0) == 0;

```

Comment:
I made this to stay completely consistent with the existing behavior for setting the complete by date for a course or assessment. Those accept zero values as blank, and there are some unit tests for it. I also agree that it seems reasonable to disallow zero values, and the Date validator handles that by saying "Enter a real date". Are we okay changing this for courses and assessments? Might be a question for @SteveJacksonSoft

Reply:
@livzorn This method's used in two endpoints - am I right in thinking one is new and the other was already there? I can see you've modified the old one's parameters - I think it's fair to change the view so it sends null days / months / years to clear the date rather than 0, especially since the two endpoints now use the same form data model. That means we have a consistent method of clearing the date. So yeah, I agree with restricting this to the case where they're all null.

Reply:
Cool, I've replaced this method with a simple check if the day month and year are all null, and adjusted the tests accordingly :)

Reply:
Nice